_tutorials/2016-09-08-grpc-in-python-for-ios-xr.md

```diff
@@ -0,0 +1,210 @@
+  
+Follow the instructions to get your topology up and running as shown below:  
+
+![app dev topo](https://xrdocs.github.io/xrdocs-images/assets/tutorial-images/app_dev_topology.png)
+
+
+If you've reached the end of the above tutorial, you should be able to issue `vagrant status` in the `vagrant-xrdocs/lxc-app-topo-bootstrap` directory to see a rtr (IOS-XR) and a devbox (Ubuntu/trusty) instance running.  
+
+**Ensure you have the latest version of IOS-XRv.**
+{: .notice--danger}
+
+## Installing gRPC on the devbox 
+
+Let's start by installing a few developer tools on the devbox.
+
+```shell
+sudo apt-get update
+sudo apt-get -y install python-dev python-pip git
+```
+
+Now that we have installed the developer tools, lets install gRPC for Python on the box.
+
+```shell
+sudo pip install grpcio
+```
+
+Thats it! gRPC for Python is installed on the devbox.
+{: .notice--success}  
+
+## Cloning the git repo
+
+Now that we have gRPC for Python installed on the devbox. We need to get the bindings associated with IOS-XR. Let's use the library that has these bindings done.
+
+Clone the gRPC for Python library here: <https://github.com/skkumaravel/iosxr-ez/tree/master/grpc>
+
+```shell
+cd ~/
+git clone https://github.com/cisco-grpc-connection-libs/ios-xr-grpc-python.git
+cd ios-xr-grpc-python
+```
+
+## Understanding the topology
+
+We need to ensure that gRPC is turned on in the devbox and take note of the port.
+
+```shell
+ssh vagrant@11.1.1.10
+```
+
+**Note the password would be vagrant**
+{: .notice--warning}
+
+From here, use `show run` to find check that gRPC is running and find the configured port.
+
+<div class="highlighter-rouge">
+<pre class="highlight">
+<code>
+RP/0/RP0/CPU0:ios#<mark>show run</mark>            
+Wed Sep  7 16:59:19.241 UTC
+Building configuration...
+!! IOS XR Configuration version = 6.1.1.19I
+!! Last configuration change at Wed Sep  7 16:18:14 2016 by UNKNOWN
+!
+telnet vrf default ipv4 server max-servers 10
+username vagrant
+ group root-lr
+ group cisco-support
+ secret 5 $1$0Lrs$AInvgKCO262qZh6hLMfis0
+!
+tpa
+ address-family ipv4
+  update-source MgmtEth0/RP0/CPU0/0
+ !
+!
+interface MgmtEth0/RP0/CPU0/0
+ ipv4 address dhcp
+!
+interface GigabitEthernet0/0/0/0
+ ipv4 address 11.1.1.10 255.255.255.0
+!
+router static
+ address-family ipv4 unicast
+  0.0.0.0/0 MgmtEth0/RP0/CPU0/0 10.0.2.2
+ !
+!
+ssh server v2
+ssh server vrf default
+<mark>grpc
+port 57777</mark>
+!
+end
+
+RP/0/RP0/CPU0:ios#
+</code>
+</pre>
+</div>
+
+We can see that gRPC is turned on, and is on port 57777.
+{: .notice--info}
+
+Let's exit out and continue getting our gRPC client working.
+
+```
+RP/0/RP0/CPU0:ios#exit
+Connection to 11.1.1.10 closed.
+vagrant@vagrant-ubuntu-trusty-64:~/ios-xr-grpc-python$
+```
+
+## Creating a Python gRPC Call
+
+There is an example call already created. We can find it in the examples folder.
+
+```shell
+cd examples
+```
+
+The file is called grpc_example.py
```

Comment:
Could you add an example for configuring the box in the tutorial? A very simple vrf or BGP config would do starting from looking at a particular YANG model that you often use - how do you create a json-ified representation of the model etc.